### PR TITLE
add PATCH endpoint /api/articles/:article_id to increase the votes on…

### PR DESCRIPTION
**Pull request by Willo84uk**
… any given article along with error handling and associated testing

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -192,12 +192,77 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+    });
+    test("400: should return a 400 error message if incorrect format is provided for article id in path", () => {
+      return request(app)
+        .patch("/api/articles/apples")
+        .send({ inc_votes: 20 })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("bad request");
+        });
+    });
```
Comment (MKD01) [inline __tests__/app.test.js:369]: Really good testing so far, but what would/should happen when a user didnt give the inc_votes property in the given object for this patch request? What would/should happen when this is also given as something other than a number?

File: endpoints.json [hunk 2]
```diff
@@ -39,6 +39,22 @@
     }
   },
 
+  "PATCH /api/articles/:article_id": {
+    "description": "increases the given article id's votecount as per the body provided in the request and returns an updated article",
+    "queries": "Article to be selected by article id within the URL",
+    "formatOfRequestBody": {"inc_votes": "NUMBER TO INCREASE BY"},
```
Comment (MKD01) [inline endpoints.json]: For this you can simply give a number as an example instead just to show a real world use case for this endpoint.

**Review by MKD01 — CHANGES_REQUESTED** [review]
Really good work so far, but your error handling tests can be added to. Once you have added some more tests please re-submit this pr

**Review by aloush-dev — APPROVED** [review]
Nice job implementing given feedback.

Merge when ready.

**Pull request merged**